pkg/tls/resolve.go

```
@@ -0,0 +1,94 @@
+	attempt := func(ctx context.Context) (bool, error) {
+		restConfig, err := libgoclient.GetKubeConfigOrInClusterConfig(kubeConfigFile, nil)
+		if err != nil {
+			lastErr = fmt.Errorf("failed to build kubeconfig: %w", err)
+		} else if configClient, cErr := configclient.NewForConfig(rest.AddUserAgent(restConfig, userAgent)); cErr != nil {
+			lastErr = fmt.Errorf("failed to create config client: %w", cErr)
+		} else {
+			resolved, lastErr = FetchAndResolve(ctx, configClient.ConfigV1())
+		}
+		if lastErr != nil {
+			klog.Warningf("failed to resolve cluster TLS security profile, will retry: %v", lastErr)
+			return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 43) but it returns nil

(nilerr)